File: Python/04-Operadores Aritméticos/Precedencia.py
#a ordem de precedência dos operadores aritméticos é
#() parenteses
#** potenciação
#* / // % multiplicação e divisão divisão inteira resto
#+ - soma e subtração
#se houver mesma precedência é resolvido da esquerda para direita

x = 5
y = 7
#parenteses tem maior precedencia
media = (x + y) / 2

#mesma precedencia é resolvido da esquerda para a direita
z = x + y - x

#raizes podem ser calculadas desta forma
raizquad = 25 ** (1/2)
raizcub = 8 ** (1/3)

print(media)
print(z)
print('raiz quadrada de 25')
print(raizquad)
print('raiz cúbica de 8')
print(raizcub)

File: C++/04-Operadores Aritméticos/Precedencia.cpp
#include<iostream>
#include<stdlib.h>

using namespace std;

int main()
{

    //A precedencia dos operadores aritméticos em C++ é
    //()
    //*/%
    //+-
    //Mesma precedencia é feito da esquerda para direita
    int x = 10;
    int y = 21;

    //parenteses tem maior precedência
    float media = (float)(x + y) / 2;
    //multiplicação ou divisão tem maior precedência que soma ou subtração
    int valor = x + y * x; //10 + 210
    //mesma precedencia é resolvido da esquerda para direita
    int valor2 = x + y - x;

    cout << media << endl;
    cout << valor << endl;
    cout << valor2 << endl;

    system("pause");
    return 0;
}
